vyper/codegen/external_call.py

```diff
@@ -118,6 +119,16 @@ def _unpack_returndata(buf, fn_type, call_kwargs, contract_address, context, exp
                 b1.resolve(make_setter(return_buf, buf, hi=add_ofst(buf, payload_bound)))
             )
     else:
+        # revert when returndatasize is not in bounds
+        # (except when return_override is provided.)
+        # make_setter checks returndatasize, except when needs_clamp is False.
```

Comment:
make_setter implicitly checks returndatasize, but it is not called when needs_clamp is False